db/view/view.cc

```diff
@@ -701,7 +706,13 @@ view_updates::get_view_rows(const partition_key& base_key, const clustering_or_s
         clustering_key ckey = clustering_key::from_range(boost::adaptors::transform(ck, view_managed_key_view_and_action::get_key_view));
         auto action = (action_column < pk.size() ? pk[action_column] : ck[action_column - pk.size()])._action;
         mutation_partition& partition = partition_for(std::move(pkey));
-        ret.push_back({&partition.clustered_row(*_view, std::move(ckey)), action});
+
+        // Skip adding the row if we already wrote a partition tombstone
+        // for this partition. This means the entire partition is deleted
+        // so we don't need to generate updates for individual rows.
+        if (!partition.partition_tombstone()) {
+            ret.push_back({&partition.clustered_row(*_view, std::move(ckey)), action});
+        }
```

Comment:
Now that I think of it, I'm not sure this condition is correct. The problem is - and I didn't tell you about before, sorry - that it's possible to both delete a partition and insert some rows into it. Cassandra and Scylla supports sending statements in batches, and if two statements refer to the same partition then they will be merged into a single mutation. For example (disclaimer: I didn't try running this batch):

```cql
BEGIN UNLOGGED BATCH
DELETE FROM ks.tbl WHERE pk = 123 USING TIMESTAMP 1000000;
INSERT INTO ks.tbl (pk, ck, v) VALUES (123, 456, 789) USING TIMESTAMP 1000001;
APPLY BATCH;
```

In this case, this will be translated to a single mutation with a partition tombstone and one clustered row.

First and foremost, please add this as a test. Then, if the optimization does not work or works but produces incorrect results then I think we should address it. Here, we should adjust the check: if there is a partition tombstone and there is no clustered row in the view update mutation, only then we can skip the check.

Reply:
I added a test and verified it produces the expected results
I added a comment explaining that the partition tombstone is the only update and therefore it should be safe to skip the rows